### PR TITLE
Update Node resource

### PR DESCRIPTION
**Pull request by omrirh**
- Update Node resource using resource generator
- Patch resources validation tests api group/version parsing

### CODE REVIEW

**Comment by myakove** [comment]
This PR delete valid code, do not merge!

**Comment by myakove** [comment]
`test_validate_resources` pass locally with your PR

```
poetry run pytest tests/test_validate_resources.py
=================================================================================== test session starts ====================================================================================
platform linux -- Python 3.12.4, pytest-8.3.2, pluggy-1.5.0
rootdir: /home/myakove/git/openshift-python-wrapper/tests
configfile: pytest.ini
collected 1 item

tests/test_validate_resources.py .                                                                                                                                                   [100%]

==================================================================================== 1 passed in 0.27s 
```

File: ocp_resources/node.py [hunk 1]
```diff
@@ -1,16 +1,102 @@
...
 from ocp_resources.resource import Resource
 
 
 class Node(Resource):
     """
-    Node object, inherited from Resource.
+    Node is a worker node in Kubernetes. Each node will have a unique identifier
+    in the cache (i.e. in etcd).
+
+    API Link: https://docs.openshift.com/container-platform/4.16/rest_api/node_apis/node-v1.html
```
Comment (rnetser) [inline ocp_resources/node.py:12]: this is a k8s resource; please add a link to the right api

File: ocp_resources/node.py [hunk 2]
```diff
@@ -35,3 +121,31 @@ def hostname(self):
     @property
     def taints(self):
         return self.instance.get("spec", {}).get("taints")
+
+    def to_dict(self) -> None:
```
Comment (rnetser) [inline ocp_resources/node.py:125]: place after init

**Review by rnetser — CHANGES_REQUESTED** [review]

File: tests/test_validate_resources.py [hunk 3]
```diff
@@ -49,7 +49,11 @@ def _process_api_type(api_type, api_value, resource_dict, cls):
 
 def _get_api_group_and_version(bodies):
     for targets in bodies:
-        api_type = targets.targets[0].id
+        if isinstance(targets, ast.AnnAssign):
```
Comment (myakove) [inline tests/test_validate_resources.py:52]: Why this change needed?

File: ocp_resources/node.py (hunk 1 above)
Comment (omrirh) [reply ocp_resources/node.py]: done

File: ocp_resources/node.py (hunk 2 above)
Comment (omrirh) [reply ocp_resources/node.py:100]: done

**Review by myakove — APPROVED** [review]

**Review by myakove — APPROVED** [review]

File: ocp_resources/node.py [hunk 4]
```diff
@@ -1,16 +1,130 @@
+# Generated using https://github.com/RedHatQE/openshift-python-wrapper/blob/main/scripts/resource/README.md
+
+from typing import Any, Dict, Optional
 from ocp_resources.resource import Resource
 
 
 class Node(Resource):
```
Comment (myakove) [inline ocp_resources/node.py:7]: Please generate the resource with the latest script code.

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocp_resources/node.py (hunk 4 above)
Comment (omrirh) [reply ocp_resources/node.py:7]: done

**Review by rnetser — APPROVED** [review]

**Review by myakove — APPROVED** [review]
